contracts/utils/ShortStrings.sol

```diff
@@ -98,4 +98,19 @@ library ShortStrings {
             return store;
         }
     }
+
+    /**
+     * @dev Return the length of a string that was encoded to `ShortString` or written to storage using {setWithFallback}.
+     */
+    function lengthWithFallback(ShortString value, string storage store) internal view returns (uint256) {
+        if (ShortString.unwrap(value) != _FALLBACK_SENTINEL) {
+            return length(value);
+        } else {
+            uint256 result;
+            assembly {
+                result := div(sload(store.slot), 2)
+            }
+            return result;
```

Comment:
Note: it would be great if we could do 

```solidity
return store.length;
``` 
or
```solidity
return StorageSlot.getStringSlot(store).value.length;
```

but neither will compile.

Reply:
I think that's just because `length` is not available for the `string` type. You should be able to do `bytes(store).length`.

By the way, the reason why length is not available is that users might expect length to be the character length of the string, but since they are UTF-8 strings the length operation is not O(1). `bytes(str).length` will return the byte length.